tests/testthat/test-write.R

```diff
@@ -163,3 +163,11 @@ test_that("hms NAs are written without padding (#930)", {
   df <- data.frame(x = hms::as.hms(c(NA, 34.234)))
   expect_equal(format_tsv(df), "x\nNA\n00:00:34.234\n")
 })
+
+test_that("More helpful error when trying to write out data frames with list columns (#938)", {
+  df <- tibble::tibble(id = seq(11))
+  df$list <- lapply(X = seq_along(df$id), FUN = rnorm, n = 3)
+  filename <- file.path(tempdir(), "test_list.csv")
+  on.exit(unlink(filename))
+  expect_error(write_csv(x = df, path = filename), "Flat files can't store the list column '")
```

Comment:
I'd just do `write_csv(df, tempfile())` — you don't need to clean up since this won't read a file.